nox/_parametrize.py

```diff
@@ -78,7 +78,7 @@ def __eq__(self, other: object) -> bool:
         elif isinstance(other, dict):
             return dict(zip(self.arg_names, self.args)) == other
 
-        raise NotImplementedError
+        return NotImplemented
```

Comment:
This doesn't seem correct. If nothing else, this doesn't type check correctly with `-> bool` return type for the function?

Reply:
@johnthagen I think that's ok. Mypy accepts `NotImplemented` as a return value for functions annotated as `-> bool`. It's in fact a little too liberal: https://github.com/python/mypy/issues/363.

That said, @fazledyn-or you probably want to update the `tests.tests__parametrize.test_param_eq_fail` test to use `assert _parametrize.Param() != "a"` instead of `pytest.raises`.

Reply:
@edgarrmondragon Thanks for the heads up.
Seems like you fixed it. Is there anything I can do ?